### PR TITLE
Chore/update "types/jest" to v29

### PR DESCRIPTION
**Pull request by PenghaiZhang**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

This PR adds the code changes required by `@types/jest` v29.

### CODE REVIEW

File: react-front-end/__tests__/tsrc/components/wizard/WizardSimpleTermSelector.test.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ import { languageStrings } from "../../../../tsrc/util/langstrings";
 describe("<WizardSimpleTermSelector/>", () => {
   // WizardSimpleTermSelector uses a debounce, so we need to be able to advanced
   // the timer to trigger a search.
-  jest.useFakeTimers("modern");
+  jest.useFakeTimers();
```
Comment (edalex-ian) [inline react-front-end/__tests__/tsrc/components/wizard/WizardSimpleTermSelector.test.tsx:29]: This is an interesting change. So you can't set a type with `useFakeTimers` any-more?

Actually, I wonder if you ever have. I went all the way back to the Jest 22 docos and it's not mentioned: https://archive.jestjs.io/docs/en/22.x/timer-mocks

But I do vaguely remember this being a thing.... :thinking:

File: react-front-end/package.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
     "@testing-library/jest-dom": "5.16.5",
     "@testing-library/react": "12.1.5",
     "@testing-library/user-event": "13.5.0",
-    "@types/jest": "27.5.2",
+    "@types/jest": "29.2.3",
```
Comment (edalex-ian) [inline react-front-end/package.json:82]: I was wondering why this is so far behind, but then I remembered #4021 which is stuck due to react stuff. :confounded:

**Review by edalex-ian — APPROVED** [review]
LGTM! Highlighted some interesting things.

Merge when happy. :+1:

File: react-front-end/__tests__/tsrc/components/wizard/WizardSimpleTermSelector.test.tsx (hunk 1 above)
Comment (PenghaiZhang) [reply react-front-end/__tests__/tsrc/components/wizard/WizardSimpleTermSelector.test.tsx:29]: You can. The new version accepts an option param which can be either `FakeTimersConfig` or `LegacyFakeTimersConfig`.

Comment (PenghaiZhang) [reply react-front-end/__tests__/tsrc/components/wizard/WizardSimpleTermSelector.test.tsx:29]: Yes we do ever have. At least at v27 we can specify `modern` or `legacy`.

File: react-front-end/package.json (hunk 2 above)
Comment (PenghaiZhang) [reply react-front-end/package.json:82]: Yeah that's the reason. :cry:

**Comment by PenghaiZhang** [comment]
I just saw this from the build output. 

`thrown: "Exceeded timeout of 5000 ms for a test.`

**Comment by edalex-ian** [comment]
> I just saw this from the build output.
> 
> `thrown: "Exceeded timeout of 5000 ms for a test.`

Interesting, it was in `MyResourcesPage.test.tsx` which had changes, but I wouldn't have expected it to cause that. (It also had some other failures.)

**Comment by PenghaiZhang** [comment]
> > I just saw this from the build output.
> > `thrown: "Exceeded timeout of 5000 ms for a test.`
> 
> Interesting, it was in `MyResourcesPage.test.tsx` which had changes, but I wouldn't have expected it to cause that. (It also had some other failures.)

I wouldn't either. This test became a bit flaky now. The second run was passed.

**Comment by edalex-ian** [comment]
Oh well, merge on and let's observe.

**Pull request merged**